tests/spec/unit/versions.spec.js

```diff
@@ -0,0 +1,91 @@
+/**
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ */
+
+var rewire = require('rewire');
+var versions = rewire('../../../bin/templates/scripts/cordova/lib/versions');
+
+// These tests can not run on windows.
+if (process.platform === 'darwin') {
+    describe('versions', function () {
+        describe('get_apple_ios_version method', () => {
+            it('should have found ios version.', (done) => {
+                let _console = versions.__get__('console');
+                let logSpy = jasmine.createSpy('logSpy');
+                versions.__set__('console', {log: logSpy});
+
+                versions.get_apple_ios_version().then(() => {
+                    expect(logSpy).not.toHaveBeenCalledWith(undefined);
+                    versions.__set__('console', _console);
+                    done();
+                });
+            });
+        });
+
+        describe('get_apple_osx_version method', () => {
+            it('should have found osx version.', (done) => {
+                let _console = versions.__get__('console');
+                let logSpy = jasmine.createSpy('logSpy');
+                versions.__set__('console', {log: logSpy});
+
+                versions.get_apple_osx_version().then(() => {
+                    expect(logSpy).not.toHaveBeenCalledWith(undefined);
+                    versions.__set__('console', _console);
+                    done();
+                });
+            });
+        });
+
+        describe('get_tool_version method', () => {
+            it('should not have found tool by name.', (done) => {
+                versions.get_tool_version('unknown').catch((error) => {
+                    expect(error).toContain('is not valid tool name');
+                    done();
+                });
+            });
+
+            it('should find xcodebuild version.', (done) => {
+                versions.get_tool_version('xcodebuild').then((version) => {
+                    expect(version).not.toBe(undefined);
+                    done();
+                });
+            });
+
+            it('should find ios-sim version.', (done) => {
```

Comment:
For some reason `versions.get_tool_version('ios-sim')`, `versions.get_tool_version('ios-deploy')`, and `versions.get_tool_version('pod')` do not work for me when I test on my local system, don't know why. Any clues?

I would also favor moving these into a separate script.

Reply:
Are these tools installed locally?  These tests can only run on a Darwin environment and with these tools installed. 

I did not mock the child process exec so it an actual test, but does not care what is the exact version.

Both locally and on the Travis CI environment has these tools are installed. I did not configure Travis CI to install them. Either the environment already came with the tools or was installed via npm install. From package.json or the .travis.yml script step.

I could mock the data, but by mocking, I would not have discovered that the Mac OSX SDK changed to macOS SDK.

Reply:
As for separate scripts, do you mean something like a bash file? In `/cordova-ios/bin` there are some individual script files that calls onto `versions.js` to fetch the value for an individual item. 

For example: 
`/cordova-ios/bin/apple_ios_version`
`/cordova-ios/bin/apple_osx_version`
`/cordova-ios/bin/apple_xcode_version`

But it is missing the `ios-sim`, `ios-deploy`, and `pod` specific executable script.

Reply:
No I meant separate .spec.js file.

I will try `npm i --save-dev ios-deploy` and see if that solves at least part of the problem.

Reply:
`npm i --save-dev ios-deploy` seems to do the trick, except for `versions.get_tool_version('pod')`; I just pushed the change to your branch

Reply:
Adding `ios-deploy` to `package.json` as a devDep will cause AppVeyor to fail. This package can only be installed and used on the Darwin platforms. 

I question if testing iOS/macOS platform on AppVeyor is actually needed. Xcode is a build requirement and Windows cant use Xcode, and AppVeyor is running on Windows, what is achieved with this test.

```
npm ERR! code EBADPLATFORM

npm ERR! notsup Unsupported platform for ios-deploy@1.9.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: ia32
```

Reply:
Also should this dep be added to the platform? 

Per documentation, it's expected that the user installed it manually and globally. 

Having it bundled with the platform might mean fewer setup steps? 
Could the globally installed ios-deploy version be affected and break when a platform major versions is released? So therefore they should become a dep?

https://cordova.apache.org/docs/en/latest/guide/platforms/ios/#installing-the-requirements